test/common-algorithms/sorting/parallel/parallel_mergesort.chpl

```diff
@@ -0,0 +1,102 @@
+
+//merges two subarrays
+//First subarray is arr[l..m]
+//Second subarray is arr[m+1..r]
+
+proc merge(arr, left:int, mid:int, right:int){
+	var i: int = 0, j: int = 0, k: int = 0;
```

Comment:
This is also one of the few parts of the language to be covered in the user's guide so far!  :)

http://chapel.cray.com/docs/master/users-guide/base/variableDeclarations.html